spec/ics-3-connection-semantics/README.md

```diff
@@ -0,0 +1,158 @@
+
+This ICS defines four subprotocols: opening handshake, header tracking, closing handshake, and closing by equivocation.
+
+##### Opening Handshake
+
+The opening handshake subprotocol serves to initialize roots of trust for two chains on each other and negotiate an agreeable connection version.
+
+This subprotocol need not be permissioned, modulo anti-spam measures.
+
+![Opening Handshake](opening_handshake.png)
+
+A correct protocol execution flows as follows:
+
+| Initiator | Datagram          | Chain |
+| --------- | ----------------- | ----- |
+| Actor     | `ConnOpenInit`    | A     |
+| Relayer   | `ConnOpenTry`     | B     |
+| Relayer   | `ConnOpenAck`     | A     |
+| Relayer   | `ConnOpenConfirm` | B     |
+
+At the end of an opening handshake between two chains implementing the subprotocol, the following properties hold:
+- Each chain has each other's correct root-of-trust as originally specified by the initiating actor.
+- The chains have agreed to a shared connection version.
+- Each chain has knowledge of and has agreed to its identifier on the other chain.
+
+##### Header Tracking
+
+The header tracking subprotocol serves to update the root of trust for an open connection.
+
+This subprotocol need not be permissioned, modulo anti-spam measures.
+
+![Header Tracking](header_tracking.png)
+
+##### Closing Handshake
+
+The closing handshake protocol serves to cleanly close a connection on two chains.
+
+This subprotocol will likely need to be permissioned to an entity who "owns" the connection on the initiating chain, such as a particular end user, smart contract, or governance mechanism.
+
+![Closing Handshake](closing_handshake.png)
+
+##### Closing by Equivocation
+
+The equivocation closing subprotocol serves to immediately close a connection if a consensus equivocation is discovered and thus prevent further packet transmission.
+
+![Closing Equivocation](closing_equivocation.png)
+
+Implementing chains may want to allow applications to register handlers to take action upon discovery of an equivocation. Further discussion is deferred to [ICS 12: Byzantine Recovery Strategies](../ics-12-byzantine-recovery-strategies).
+
+### Backwards Compatibility
+
+Not applicable.
+
+### Forwards Compatibility
+
+Once a connection has been established and a version negotiated, future version updates can be negotiated per [ICS 6: Connection & Channel Versioning](../spec/ics-6-connection-channel-versioning). The root of trust can only be updated as allowed by the `updateRootOfTrust` function defined by the consensus protocol chosen when the connection is established.
```

Comment:
With a open structure (e.g. protobuf structures are open in the sense that new fields can be defined later), additional functionality can be added at a later time, so it may not be necessary to define how version upgrading works initially.